src/main/java/org/opensrp/web/rest/EventResource.java

```diff
@@ -441,44 +443,54 @@ private void searchMissingClients(List<String> clientIds, List<Client> clients,
 			if (syncData.has("clients")) {
 				ArrayList<Client> clients = gson.fromJson(Utils.getStringFromJSON(syncData, "clients"),
 				    new TypeToken<ArrayList<Client>>() {}.getType());
+				logger.error("[EVENT_RESOURCE] " + clients.size() +  " clients submitted by user " + username);
+				long currentTime = System.currentTimeMillis();
 				for (Client client : clients) {
 					try {
 						clientService.addorUpdate(client);
 					}
 					catch (Exception e) {
-						logger.error(
-						    "Client" + client.getBaseEntityId() == null ? "" : client.getBaseEntityId() + " failed to sync",
-						    e);
+						client.getBaseEntityId();
+						logger.error( "Client" + client.getBaseEntityId() + " failed to sync",e);
 						failedClientsIds.add(client.getId());
 					}
 				}
+				logger.error("[EVENT_RESOURCE] processed " + clients.size() + " clients in"
+						+ ((System.currentTimeMillis() - currentTime) / 1000) + " seconds");
 				
 			}
+
 			if (syncData.has("events")) {
 				ArrayList<Event> events = gson.fromJson(Utils.getStringFromJSON(syncData, "events"),
 				    new TypeToken<ArrayList<Event>>() {}.getType());
+				logger.error("[EVENT_RESOURCE] " + events.size() +  " events submitted by user " + username);
+				long currentTime = System.currentTimeMillis();
 				for (Event event : events) {
 					try {
 						event = eventService.processOutOfArea(event);
-						eventService.addorUpdateEvent(event, RestUtils.currentUser(authentication).getUsername());

```

Comment:
Could we add a log between these two line?

Reply:
:+1: